src/datepicker/datepicker.component.ts

```diff
@@ -213,13 +218,25 @@ export class DatePicker implements OnDestroy, OnChanges, AfterViewChecked {
 		}
 	}
 
+	ngAfterViewInit() {
+		this.visibilitySubscription = this.elementService
+			.visibility(this.elementRef.nativeElement, this.elementRef.nativeElement)
+			.subscribe(value => {
+				if (this.isFlatpickrLoaded()) {
```

Comment:
I think you'll want to do something like 
```suggestion
				if (this.isFlatpickrLoaded() && this.flatpickrInstance.isOpen) {
```

I'm not 100% sure of the property names though 👍